### PR TITLE
#162380276 view login page

### PR DESCRIPTION
**Pull request by augustineezinwa**
#### What does this PR do?
- This PR creates the view for the login page.

#### Description of Task to be completed?
- Implemented all feedback received. Improved the mobile login view
#### How should this be manually tested?
- `npm test`
#### Any background context you want to provide?
- None
#### What are the relevant pivotal tracker stories?
- #162380276
#### Screenshots (if appropriate)
screen shot 2018-12-07 at 9 09 34 pm
screen shot 2018-12-07 at 9 09 21 pm

#### Link to review app: https://valinor-ah-frontend-stag-pr-16.herokuapp.com/login

### CODE REVIEW

**Comment by sulenchy** [comment]
Looking at the mobile view in the mockup at https://www.figma.com/proto/YGLO393YeI9xgnkSS7iMkuq4/Author's-Haven-Mockups?node-id=555%3A1817&scaling=scale-down, I can see the mobile view you have needs to be modified. A user who wants to log in on mobile will have to scroll down to the bottom. This is not in conformity with what we have in the mockup. You can modify the paddings and margin.

File: src/components/login/LoginForm.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+const LoginForm = () => (
+  <form>
+    <div className="row mt-5 text-center">
+      <div className="col-md-8 offset-md-2">
+        <div className="input-group mb-3">
+          <div className="input-group-prepend">
+            <span className="input-group-text transparent email-addon" id="basic-addon1"><i className="far fa-envelope" /></span>
+          </div>
+          <input type="text" className="form-control email-input" placeholder="EMAIL ADDRESS" aria-label="Username" aria-describedby="basic-addon1" />
```
Comment (mbilesanmi) [inline src/components/login/LoginForm.js]: Should this be a textType?

File: src/components/login/Button.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
+import React from 'react';
```
Comment (mbilesanmi) [inline src/components/login/Button.js]: Is this file really necessary?

File: src/components/login/SocialButton.js [hunk 3]
```diff
@@ -0,0 +1,18 @@
+import React from 'react';
```
Comment (mbilesanmi) [inline src/components/login/SocialButton.js]: Is this file also required?

**Comment by mbilesanmi** [comment]
The design seems a bit off on mobile though.

Good work so far.

File: src/components/login/SocialButton.js (hunk 3 above)
Comment (augustineezinwa) [reply src/components/login/SocialButton.js]: it is necessary because there are places where it can be reused. So I decided to create a single instance and reused it at these different places by changing the props.

File: src/components/login/LoginForm.js (hunk 1 above)
Comment (augustineezinwa) [reply src/components/login/LoginForm.js]: it should be 'email'. Thanks for the feedback

File: public/assets/styles/login.scss [hunk 4]
```diff
@@ -0,0 +1,121 @@
+@import url('https://fonts.googleapis.com/css?family=Zilla+Slab');
+@import url('https://fonts.googleapis.com/css?family=Open Sans');
```
Comment (KvNGCzA) [inline public/assets/styles/login.scss]: Nice, but these two import links should be one.

File: public/assets/styles/login.scss [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+  @extend %style-button;
+  background-color: $facebook-color;
+}
+
+.twitter-btn{
+  @extend %style-button;
+  background-color: $twitter-color;
+}
+
+.twitter-btn:hover{
```
Comment (KvNGCzA) [inline public/assets/styles/login.scss]: this should be nested inside 
.twitter-button{
 &:hover {
 }
}

File: public/assets/styles/login.scss [hunk 6]
```diff
@@ -0,0 +1,121 @@
...
+  background-color: $twitter-color;
+}
+
+.facebook-btn:hover{
+  background-color: $facebook-color;
```
Comment (KvNGCzA) [inline public/assets/styles/login.scss]: same as my comment on line 65

File: public/assets/styles/login.scss [hunk 7]
```diff
@@ -0,0 +1,121 @@
...
+}
+
+.google-btn:hover{
```
Comment (KvNGCzA) [inline public/assets/styles/login.scss]: same as my comment on line 65

File: public/assets/styles/login.scss [hunk 8]
```diff
@@ -0,0 +1,121 @@
...
+
+.email-input{
+  @extend %input-style;
+  background-color: transparent;
+}
+
+.email-input:focus {
+  @extend %input-style;
+  background-color: transparent;
+  box-shadow: none;
```
Comment (KvNGCzA) [inline public/assets/styles/login.scss]: does this element come with a default box-shadow?? if it doesn't, then this is not needed

File: src/components/login/LoginPage.js [hunk 9]
```diff
@@ -1,11 +1,19 @@
 import React from 'react';
 
 import Header from '../common/Header';
+import LoginSection from './LoginSection';
+import WelcomeSection from './WelcomeSection';
+import '../../../public/assets/styles/login.scss';
```
Comment (KvNGCzA) [inline src/components/login/LoginPage.js]: this is not how we agreed to import our scss files, refer to [this](https://github.com/andela/valinor-ah-frontend/blob/4d70aebf029554c024a3e0e1fa0117df12433849/public/styles/index.scss#L5)

File: src/index.html [hunk 10]
```diff
@@ -4,6 +4,8 @@
   <meta charset="UTF-8" />
   <meta name="viewport" content="width=device-width, initial-scale=1.0">
   <meta http-equiv="X-UA-Compatible" content="ie=edge">
+  <link rel="stylesheet" href="https://use.fontawesome.com/releases/v5.5.0/css/all.css" integrity="sha384-B4dIYHKNBt8Bc12p+WXckhzcICo0wtJAoU8YZTY5qE0Id1GSseTk6S+L3BlXeVIU" crossorigin="anonymous">
+  <link href='https://fonts.googleapis.com/css?family=Zilla Slab' rel='stylesheet'>
```
Comment (KvNGCzA) [inline src/index.html]: This line is already included in the index.scss file

**Review by KvNGCzA — CHANGES_REQUESTED** [review]

File: public/assets/styles/login.scss [hunk 11]
```diff
@@ -0,0 +1,121 @@
+@import url('https://fonts.googleapis.com/css?family=Zilla+Slab');
+@import url('https://fonts.googleapis.com/css?family=Open Sans');
+
+$background-color: #f9f4f4;
+$facebook-color: #3b5998;
+$twitter-color: #00aced;
+$google-color: #f44d4d;
```
Comment (KvNGCzA) [inline public/assets/styles/login.scss]: all variables should be in a variables.scss file

File: public/assets/styles/login.scss [hunk 12]
```diff
@@ -0,0 +1,121 @@
...
+  border: 0;
+  border-bottom: 1px solid $line-color;
+  border-radius: 0;
+}
+
+.login-form{
+  background-color: $background-color;
+  border-bottom-left-radius: 10px;
+  border-top-left-radius: 10px;
+  padding: 80px 40px;
```
Comment (KvNGCzA) [inline public/assets/styles/login.scss]: in line with the teams conventions, we are to use em and not px for dimensions. Please resolve this.

File: public/assets/styles/login.scss (hunk 5 above)
Comment (augustineezinwa) [reply public/assets/styles/login.scss]: Thanks for the feedback. I have implemented them

File: public/assets/styles/login.scss (hunk 12 above)
Comment (augustineezinwa) [reply public/assets/styles/login.scss]: Done

File: public/assets/styles/index.scss [hunk 13]
```diff
@@ -0,0 +1,52 @@
+@import url(https://fonts.googleapis.com/css?family=Zilla+Slab:300,300i,400,500,600,700%7COpen+Sans:300,400,600,700,800);
+@import url('https://fonts.googleapis.com/css?family=Open Sans');
```
Comment (sulenchy) [inline public/assets/styles/index.scss]: This line of code should not be there again because the first line includes both `Zilla slab` and `Open Sans`. Please remove it.

Comment (augustineezinwa) [reply public/assets/styles/index.scss]: Ok. Thanks for the feedback. Let me implement that.

File: public/assets/styles/login.scss (hunk 4 above)
Comment (augustineezinwa) [reply public/assets/styles/login.scss]: @KvNGCzA Thanks. its done

File: public/assets/styles/index.scss (hunk 13 above)
Comment (augustineezinwa) [reply public/assets/styles/index.scss]: @sulenchy done.

File: public/assets/styles/index.scss [hunk 14]
```diff
@@ -0,0 +1,51 @@
+@import url(https://fonts.googleapis.com/css?family=Zilla+Slab:300,300i,400,500,600,700%7COpen+Sans:300,400,600,700,800);
+@import './variables';
+@import './login';
```
Comment (KvNGCzA) [inline public/assets/styles/index.scss]: why have you removed the mixins import in this file? Please put it back in so that it does not affect others who are using mixins. Also, include a new line after line 1 for clarity. Thanks.

File: public/assets/styles/login.scss [hunk 15]
```diff
@@ -0,0 +1,255 @@
...
+}
+
+@media only screen and (max-width: 768px) {
+  .account-question {
+    font-size: 90%;
+    margin-top: 80%;
+  }
+}
+
+@media only screen and (max-width: 767px) {
```
Comment (KvNGCzA) [inline public/assets/styles/login.scss]: can't all the styles under this media query also fall under the media query on line 157? I believe it can.

File: public/assets/styles/login.scss [hunk 16]
```diff
@@ -0,0 +1,255 @@
...
+  margin-top: 3rem;
+}
+
+.welcome-text {
+  font-size: 2.25em;
+  font-weight: bold;
+  -webkit-text-stroke: 0.03em $white;
+}
+
+h2, h5, h4{
```
Comment (KvNGCzA) [inline public/assets/styles/login.scss]: please make this styling specific to the container you want to use it in. This is currently a general style and will affect other peoples work.

File: public/assets/styles/login.scss [hunk 17]
```diff
@@ -0,0 +1,255 @@
...
+  font-family: 'Zilla Slab';
+}
+
+p{
+  font-family: 'Open Sans';
```
Comment (KvNGCzA) [inline public/assets/styles/login.scss]: same as line 127. Also, there is no need to set the font-family here. We already did this on line 7 in the index.scss file. 'Zilla Slab' is the only font-family you need to set anywhere.

File: public/assets/styles/login.scss (hunk 15 above)
Comment (augustineezinwa) [reply public/assets/styles/login.scss]: No, it cant. I wanted a particular style at that point. That was why I included that query.

File: public/assets/styles/login.scss (hunk 17 above)
Comment (augustineezinwa) [reply public/assets/styles/login.scss]: Feedback well taken.

File: public/assets/styles/login.scss (hunk 16 above)
Comment (augustineezinwa) [reply public/assets/styles/login.scss]: Feedback well taken

File: public/assets/styles/login.scss (hunk 15 above)
Comment (KvNGCzA) [reply public/assets/styles/login.scss]: The difference between the two media queries is 1px. I think this can be done. And whatever problems you may be having with your margins and font-size can be fixed by using em's for your dimensions and not percentages.

File: public/assets/styles/index.scss (hunk 14 above)
Comment (augustineezinwa) [reply public/assets/styles/index.scss]: Cool. The latest pull I had from develop never had that mixins file. My PR doesn't make use of any mixins file. So whosoever is to rebase after any PR gets merged can bring all needed files together.

Comment (augustineezinwa) [reply public/assets/styles/index.scss]: For the line 1, let me add that.

Comment (KvNGCzA) [reply public/assets/styles/index.scss]: We sent the same index.scss file to everyone to use via slack on valinor-general. This file is not based on develop. You are the only one who removed it. Whether you are using it or not, it should be in there. Please put it back in.

File: public/assets/styles/variables.scss [hunk 18]
```diff
@@ -0,0 +1,20 @@
+// color variables
+$nav-red: #ff0000;
+$nav-item-black: #3a3a3a;
+$color-accent: #007fff;
+$color-bg: #fff;
```
Comment (mbilesanmi) [inline public/assets/styles/variables.scss]: Always write out the colours in full. `#ffffff` not `#fff`

File: src/App.js [hunk 19]
```diff
@@ -6,6 +6,7 @@ import store from './store/store';
 import NotFound from './components/404/NotFound';
 
 const App = () => (
+
```
Comment (mbilesanmi) [inline src/App.js]: Why add this new line?

File: public/assets/styles/variables.scss (hunk 18 above)
Comment (augustineezinwa) [reply public/assets/styles/variables.scss]: Feedback well received

File: src/components/login/LoginPage.js [hunk 20]
```diff
@@ -1,11 +1,81 @@
-import React from 'react';
 
+import React from 'react';
+import emaiIcon from '../../../public/assets/icons/email-icon.svg';
```
Comment (mbilesanmi) [inline src/components/login/LoginPage.js]:
```suggestion
import emailIcon from '../../../public/assets/icons/email-icon.svg';
```

File: src/components/login/LoginPage.js [hunk 21]
```diff
@@ -1,11 +1,81 @@
...
+          <div className="row welcome-banner align-items-center">
+            <div className="col-md-12">
+              <h2 className="welcome-text">Welcome back!</h2>
+              <p className="text-center welcome-body">To keep sharing your exciting Ideas please log in into your account</p>
+            </div>
+          </div>
+
+          <div className="row login-banner">
+            <div className="col-md-12">
+              <p className="account-question">don't have an account yet?</p>
```
Comment (mbilesanmi) [inline src/components/login/LoginPage.js]: use `don&apos;t` instead of `dont't`

**Review by mbilesanmi — CHANGES_REQUESTED** [review]

File: src/components/login/LoginPage.js (hunk 21 above)
Comment (augustineezinwa) [reply src/components/login/LoginPage.js]: Feedback well received.

File: package.json [hunk 22]
```diff
@@ -29,7 +29,17 @@
   },
   "homepage": "https://github.com/andela/valinor-ah-frontend#readme",
   "devDependencies": {
+    "@babel/core": "^7.1.6",
```
Comment (mbilesanmi) [inline package.json]: Are all of these changes required?

File: public/styles/login.scss [hunk 23]
```diff
@@ -0,0 +1,250 @@
+%style-button {
```
Comment (mbilesanmi) [inline public/styles/login.scss:1]: Most of these styling is similar to the signup style. Why not extend that?

File: public/assets/icons/email-icon.svg [hunk 24]
```diff
@@ -0,0 +1,9 @@
+<svg width="19" height="18" viewBox="0 0 19 18" fill="none" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
```
Comment (mbilesanmi) [inline public/assets/icons/email-icon.svg]: This file exists as a slightly different name, if I'm correct

File: public/styles/login.scss (hunk 23 above)
Comment (augustineezinwa) [reply public/styles/login.scss:1]: Feedback well received

File: public/assets/icons/email-icon.svg (hunk 24 above)
Comment (augustineezinwa) [reply public/assets/icons/email-icon.svg]: Yes, you are correct. feedback well received

File: src/components/login/LoginPage.js [hunk 25]
```diff
@@ -1,11 +1,83 @@
...
+            </form>
+            {/* End of Email input */}
+
+          </div>
+
+
+        </div>
+      </div>
+    </div>
   </div>
```
Comment (KvNGCzA) [inline src/components/login/LoginPage.js:84]: You can now input the footer here after rebasing.

File: public/styles/login.scss [hunk 26]
```diff
@@ -0,0 +1,244 @@
...
+  }
+
+  .account-question {
+    font-size: 80%;
+    margin-top: 12%;
+  }
+
+  .welcome-text {
+    font-size: 135%;
+    font-weight: 900;
```
Comment (KvNGCzA) [inline public/styles/login.scss]: The biggest font weight we have is 700, I'm sure if you change this to 700 it will return the same result.

File: public/styles/login.scss [hunk 27]
```diff
@@ -0,0 +1,244 @@
...
+    margin-top: 4%;
+  }
+
+  .account-question {
+    font-size: 80%;
+    margin-top: 12%;
+  }
+
+  .welcome-text {
+    font-size: 135%;
```
Comment (KvNGCzA) [inline public/styles/login.scss]: can I ask why you are using percentages for your font-sizes?

File: src/App.js (hunk 19 above)
Comment (augustineezinwa) [reply src/App.js]: I have fixed this

File: public/styles/login.scss (hunk 26 above)
Comment (augustineezinwa) [reply public/styles/login.scss]: Feedback received

File: src/components/login/LoginPage.js (hunk 25 above)
Comment (augustineezinwa) [reply src/components/login/LoginPage.js:84]: Feedback well received

File: package.json (hunk 22 above)
Comment (mbilesanmi) [reply package.json]: @augustineezinwa This doesn't seem to have been done or answered.

File: public/styles/login.scss (hunk 27 above)
Comment (mbilesanmi) [reply public/styles/login.scss]: Let's stick to using `em` or` rem`, please.

File: public/styles/login.scss (hunk 23 above)
Comment (mbilesanmi) [reply public/styles/login.scss:1]: Has this been done?

File: public/styles/login.scss (hunk 27 above)
Comment (augustineezinwa) [reply public/styles/login.scss]: Ok.

File: package.json (hunk 22 above)
Comment (augustineezinwa) [reply package.json]: Let me fix this

File: src/components/login/LoginPage.js (hunk 25 above)
Comment (augustineezinwa) [reply src/components/login/LoginPage.js:84]: The footer file in develop is empty.

File: package.json (hunk 22 above)
Comment (augustineezinwa) [reply package.json]: @mbilesanmi  done

File: public/styles/login.scss (hunk 27 above)
Comment (augustineezinwa) [reply public/styles/login.scss]: @mbilesanmi done

Comment (augustineezinwa) [reply public/styles/login.scss]: @mbilesanmi  done

**Review by mbilesanmi — APPROVED** [review]

**Comment by mbilesanmi** [comment]
@KvNGCzA @Darthrighteous @mrSerious Please review this PR ASAP.....

**Review by tersoo-atsen — APPROVED** [review]

**Comment by darthrighteous** [comment]
screen shot 2018-12-12 at 9 34 17 am

1. The sign up and login pages have different heights
2. The text above the `signup` and `login` buttons (the ones to switch the page) are different

Fix those and I think this is ready to get merged.

**Comment by mbilesanmi** [comment]
1. For the signup page, should the social buttons be `login` or `signup`?
2. Please use HTML validations on the forms. A simple `required` would do a lot in some cases.
3. When Clicking `Login` on signup and vice versa, should it trigger a page reload.
4. Please rebase ASAP...

@augustineezinwa

**Comment by augustineezinwa** [comment]
> 1. For the signup page, should the social buttons be `login` or `signup`?
> 2. Please use HTML validations on the forms. A simple `required` would do a lot in some cases.
> 3. When Clicking `Login` on signup and vice versa, should it trigger a page reload.
> 4. Please rebase ASAP...
> 
> @augustineezinwa

@mbilesanmi for the social buttons; it should be `signup`
 Feedback received.
For question 4. I am not so sure how to implement that. They are on different pages. But let me find out.

**Comment by augustineezinwa** [comment]
@mbilesanmi done

**Comment by mbilesanmi** [comment]
@augustineezinwa Have you rebased?
